codegen/src/api/mod.rs

```diff
@@ -621,7 +624,12 @@ impl RuntimeGenerator {
 
                 /// check whether the Client you are using is aligned with the statically generated codegen.
                 pub fn validate_codegen<T: #crate_path::Config, C: #crate_path::client::OfflineClientT<T>>(client: &C) -> Result<(), #crate_path::error::MetadataError> {
-                    let runtime_metadata_hash = client.metadata().hasher().only_these_pallets(&PALLETS).hash();
+                    static RUNTIME_APIS: [&str; #runtime_api_names_len] = [ #(#runtime_api_names,)* ];
```

Comment:
For consistency I'd probably move this up next to the PALLETS static on line 544 (you could also move the PALLETS static down here; I wonder whether it's useful to publically expose these or not).